Updated README (#171)
Azure/reliable-web-app-pattern-java: README.md

## Prerequisites

We recommend that you use a DevContainer to deploy this application.  The requirements are as follows:

- [Azure Subscription](https://azure.microsoft.com/pricing/member-offers/msdn-benefits-details/).
- [Visual Studio Code](https://code.visualstudio.com/).
- [Docker Desktop](https://www.docker.com/get-started/).

> - [Permissions to register an application](https://learn.microsoft.com/azure/active-directory/develop/quickstart-register-app).

Reviewer: ```suggestion
- [Permissions to register an application in Azure AD](https://learn.microsoft.com/azure/active-directory/develop/quickstart-register-app).
```

- Visual Studio Code [Dev Containers extension](https://marketplace.visualstudio.com/items?itemName=ms-vscode-remote.remote-containers).

If you do not wish to use a DevContainer, please refer to the [prerequisites](prerequisites.md) for detailed information on how to set up your development system to build, run, and deploy the application.

> **WINDOWS**
> If you are using Windows, you must enable long paths.  Do the following as a local Administrator:
>
> - Open the Registry Editor, navigate to `HKLM\SYSTEM\CurrentControlSet\Control\FileSystem` and set DWORD `LongPathsEnabled` to 1.
> - Run `git config --system core.longpaths true`

## Steps to deploy the reference implementation

The short version (assumes you are already logged into Azure through the Azure CLI and Azure Developer CLI, and that a suitable subscription is selected):

```shell
git clone https://github.com/Azure/reliable-web-app-pattern-java.git
cd reliable-web-app-pattern-java
azd config set alpha.terraform on
azd env new eap-javarwa
azd env set DATABASE_PASSWORD "AV@lidPa33word"
azd env set APP_ENVIRONMENT prod
azd env set AZURE_LOCATION westus3
azd env set AZURE_LOCATION2 eastus

> azd up

Reviewer: This approach appears to demonstrate multi-regional provision and single region deploy. The azd up command is unaware of the second resource group.

I think we should choose to deploy the simple version as single region or add the following

> SECONDARY_RESOURCE_GROUP=$(azd env get-values --output json | jq -r .secondary_resource_group)
> azd env set AZURE_RESOURCE_GROUP $SECONDARY_RESOURCE_GROUP
> azd deploy

Author: Fixed.

```

> The following walk-through assumes you are using a DevContainer inside Visual Studio Code and walks you through executin

Reviewer: ```suggestion
The following walk-through assumes you are using a DevContainer inside Visual Studio Code and walks you through executing the above sequence.
```

> The following walk-through assumes you are using a DevContainer inside Visual Studio Code and walks you through executin

Reviewer: I think it should be `Dev Container` as two words - https://learn.microsoft.com/en-us/shows/beginners-series-to-dev-containers/

Author: Fixed throughout.

> g the above sequence.

Reviewer: ```suggestion

```

```shell
code .
```

Once Visual Studio Code is launched, you should see a popup allowing you to click on the button **Reopen in Container**.

![Reopen in Container](docs/assets/vscode-reopen-in-container.png)

If you don't see the popup, open the Visual Studio Code Command Palette to execute the command. There are three ways to open the command palette:

- For Mac users, use the keyboard shortcut ⇧⌘P
- For Windows and Linux users, use Ctrl+Shift+P
- From the Visual Studio Code top menu, navigate to View -> Command Palette.

Once the command palette is open, search for `Dev Containers: Rebuild and Reopen in Container`.

![WSL Ubuntu](docs/assets/vscode-reopen-in-container-command.png)

### 3. Log in to Azure

Before deploying, you must be authenticated to Azure and have the appropriate subscription selected.  To authenticate:

```shell
az login --scope https://graph.microsoft.com//.default
azd auth login
```

Each command will open a browser allowing you to authenticate.  To list the subscriptions you have access to:

```shell
az account list
```

To set the active subscription:

```shell

> az accout set --subscription "<your-subscription-id>"

Reviewer: ```suggestion
az account set --subscription "<your-subscription-id>"
```

> az accout set --subscription "<your-subscription-id>"

Reviewer: Found that when I applied copy and paste as recommended, I had to go back to the portal and get my subscriptionId again.

Would like to see "<your-subscription-id>" assigned to a variable or some other shorthand approach.

Author: Fixed.
